test/ctl-e2e-test.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+load_settings () {
+    source "$(dirname "$(realpath "$0")")/e2e_defaults.env"
+}
+
+load_settings
+
+# find a suitable browser for e2e-tests
+find_browser () {
+    which chromium google-chrome | head -n1
+}
+
+# find a suitable temp directory for e2e-tests. snaps don't work
+# in $TMPDIR, because of lacking read access!
+temp_base () {
+    BROWSER=$(find_browser)
+    if [[ "$BROWSER" == *"snap"* ]]; then
+        echo ./tmp
+    else
+        eche $TMPDIR
+    fi
+}
+
+unique_temp_dir () {
+    echo "$(temp_base)/$(mktemp -du e2e.XXXXXXX)"
+}
+
+run_tests () {
+    UNIQUE_TEMP_DIR=$(unique_temp_dir)
+    mkdir -p $UNIQUE_TEMP_DIR
+    unzip $NAMI_CRX -d $UNIQUE_TEMP_DIR/nami > /dev/zero
+    tar xzf $NAMI_SETTINGS
+    unzip $GERO_CRX -d $UNIQUE_TEMP_DIR/gero > /dev/zero
+    tar xzf $GERO_SETTINGS
+    unzip $FLINT_CRX -d $UNIQUE_TEMP_DIR/flint > /dev/zero
+    tar xzf $FLINT_SETTINGS
+    unzip $LODE_CRX -d $UNIQUE_TEMP_DIR/lode > /dev/zero
+    tar xzf $LODE_SETTINGS
+    rm -f $CHROME_PROFILE/SingletonLock
+    spago test --main Test.E2E -a "E2ETest \
+          --lode-dir $UNIQUE_TEMP_DIR/lode \
+          --lode-password $LODE_PASSWORD \
+          $* --chrome-exe $(find_browser)" || rm -Rf $UNIQUE_TEMP_DIR
```

Comment:
The `$*` refers to arguments, but no arguments are passed when it's called later on in the script